index.bs

```diff
@@ -606,6 +701,39 @@ SessionResult = (StatusResult)
 
 ```
 
+<div algorithm>
+
+To <dfn lt="updating the event map">update the event map</dfn>, given a |list of
+event names|, a |list of contexts|, and a |target status|:
+
+  1. Let |events| be an empty set.
+
+    1. For each entry |event name| in the |list of event names|,
```

Comment:
The indentation of this and the following is one level too much, resulting in the algorithm having just one main step :)